odl/discr/discr_mappings.py

```diff
@@ -687,42 +790,185 @@ def _evaluate(self, indices, norm_distances, out=None):
             return self.values[idx_res]
 
 
-class _NearestMeshgridInterpolator(_MeshgridInterpolator,
-                                   _NearestPointwiseInterpolator):
+def _compute_nearest_weights_edge(idcs, ndist, variant):
+    """Helper for nearest interpolation mimicing the linear case."""
+    # Get out-of-bounds indices from the norm_distances. Negative
+    # means "too low", larger than or equal to 1 means "too high"
+    lo = np.where(ndist < 0)
+    hi = np.where(ndist > 1)
+    ones = np.ones_like(ndist)
+    zeros = np.zeros_like(ndist)
+
+    # For "too low" nodes, the lower neighbor gets weight zero;
+    # "too high" gets 1.
+    if variant == 'left':
+        w_lo = np.where(ndist <= 0.5, ones, zeros)
+    else:
+        w_lo = np.where(ndist < 0.5, ones, zeros)
+
+    w_lo[lo] = 0
+    w_lo[hi] = 1
+
+    # For "too high" nodes, the upper neighbor gets weight zero;
+    # "too low" gets 1.
+    if variant == 'left':
```

Comment:
Not so sure about this. Numerical stability tells me that this will have some real issues and since floating point behaviour is not 100% defined, we could even get implementation dependent behavior. I cant think of many cases where we could guarantee this feature to work as people expect it to:

```python
>>> np.array([0.2 + 0.1, 0.3]) <= 0.3
array([ False, True], dtype=bool)
```